lib/workflow/stores/mongo.js

```diff
@@ -883,5 +883,23 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
         return waterline.graphobjects.findOne(query);
     };
 
+    /**
+     * Finds the child graph of a runGraphJob by the job's task id/the child
+     * graph's parentTaskId
+     * @param {String} parentTaskId - The task id of a runGraphTask to match
+     * against a graph's context._parent.taskId field
+     * @returns {Promise} a promise for the graph with the given context._parent.taskId
+     * @memberOf store
+     */
+    exports.findChildGraph = function(parentTaskId) {
+        assert.uuid(parentTaskId, "Parent task ID should be a UUID");
+
+        var query = {
+            "context._parent.taskId": parentTaskId
```

Comment:
Right, but with that in mind, any thoughts on how to make that easier upfront rather than planning to worry about it later :smiley: ? https://github.com/RackHD/RackHD/wiki/(optional-configurable)-Persistence-Layers-for-RackHD

Do you want another model to define this relationship rather than stashing it in the context?  Is there another query based on something in the base of the model that can be used to reduce the data set to something that is manageable for post-query processing?  Should the graph object model be modified to accommodate what you are trying to do here?